Packet++/header/FtpLayer.h

```diff
@@ -0,0 +1,458 @@
+			USER_OK_NEED_PASS_CHALLENGE = 336,
+			/// Requested file action pending further information
+			FILE_PENDING_ACTION = 350,
+			/// Service not available, closing control connection
+			SERVICE_NOT_AVAILABLE = 421,
+			/// Can't open data connection
+			CANT_OPEN_DATA_CONNECTION = 425,
+			/// Connection closed; transfer aborted
+			CONNECTION_CLOSED = 426,
+			/// Need some unavailable resource to process security
+			NEED_UNAVAILABLE_RESOURCE_TO_SEC = 431,
+			/// Requested file action not taken
+			REQ_FILE_ACTION_NOT_TAKEN = 450,
+			/// Requested action aborted: local error in processing
+			REQ_ACTION_ABORTED = 451,
+			/// Requested action not taken. Insufficient storage space in system
+			REQ_ACTION_NOT_TAKEN = 452,
+			/// Syntax error, command unrecognized
+			SYNTAX_ERROR_COMMAND_UNRECOGNIZED = 500,
+			/// Syntax error in parameters or arguments
+			SYNTAX_ERROR_PARAMETER_OR_ARGUMENT = 501,
+			/// Command not implemented
+			COMMAND_NOT_IMPLEMENTED = 502,
+			/// Bad sequence of commands
+			BAD_SEQUENCE_COMMANDS = 503,
+			/// Command not implemented for that parameter
+			COMMAND_NOT_IMPLEMENTED_FOR_PARAMETER = 504,
+			/// Network protocol not supported
+			NETWORK_PROTOCOL_NOT_SUPPORTED = 522,
+			/// Not logged in
+			NOT_LOGGED_IN = 530,
+			/// Need account for storing files
+			NEED_ACCOUNT_FOR_STORE_FILE = 532,
+			/// Command protection level denied for policy reasons
+			COMMAND_PROTECTION_DENIED = 533,
+			/// Request denied for policy reasons
+			REQUEST_DENIED = 534,
+			/// Failed security check (hash, sequence, etc)
+			FAILED_SEC_CHECK = 535,
+			/// Requested PROT level not supported by mechanism
+			REQ_PROT_LEVEL_NOT_SUPPORTED = 536,
+			/// Command protection level not supported by security mechanism
+			COMMAND_PROTECTION_LEVEL_NOT_SUPPORTED = 537,
+			/// Requested action not taken: File unavailable
+			FILE_UNAVAILABLE = 550,
+			/// Requested action aborted: page type unknown
+			PAGE_TYPE_UNKNOWN = 551,
+			/// Requested file action aborted: Exceeded storage allocation
+			EXCEED_STORAGE_ALLOCATION = 552,
+			/// Requested action not taken: File name not allowed
+			FILENAME_NOT_ALLOWED = 553,
+			/// Integrity protected reply
+			INTEGRITY_PROTECTED = 631,
+			/// Confidentiality and integrity protected reply
+			CONFIDENTIALITY_AND_INTEGRITY_PROTECTED = 632,
+			/// Confidentiality protected reply
+			CONFIDENTIALITY_PROTECTED = 633
+		};
+
+		/** A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		FtpResponseLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet) : FtpLayer(data, dataLen, prevLayer, packet) { m_Protocol = FTP; };
+
+		/**
+		 * Empty c'tor
+		 */
+		FtpResponseLayer();
```

Comment:
Same here: this c'tor shouldn't be empty and accept the status code and (optional) argument, like:
```cpp
pcpp::FtpResponseLayer ftpResLayer(pcpp::FtpResponseLayer::SYSTEM_STATUS, "bla bla");
```